protocols/gossipsub/src/behaviour.rs

```diff
@@ -741,11 +776,40 @@ where
                         return Ok(false);
                     }
                 };
-                self.forward_msg(msg_id, raw_message, Some(propagation_source))?;
+                self.forward_msg(msg_id, raw_message.clone(), Some(propagation_source))?;
+                self.mesh_slot_data
+                    .entry(raw_message.topic.clone())
+                    .or_insert_with(|| MeshSlotData::new(raw_message.topic.clone()))
+                    .increment_message_metric(
+                        propagation_source,
+                        SlotMessageMetric::MessagesValidated,
+                    );
                 return Ok(true);
             }
-            MessageAcceptance::Reject => RejectReason::ValidationFailed,
-            MessageAcceptance::Ignore => RejectReason::ValidationIgnored,
+            MessageAcceptance::Reject => {
+                if let Some(raw_message) = self.mcache.validate(msg_id) {
```

Comment:
The user here is rejecting the message. Why are we validating it here? Won't this mean that we will then gossip messages that have been rejected at the user level?